packages/web3/test/integration/main_web3.test.ts

```diff
@@ -75,15 +75,34 @@ describe('Web3 instance', () => {
 		if (getSystemTestProvider().startsWith('ws')) {
 			// make sure we try to close the connection after it is established
 			if (
-				web3.provider &&
+				web3?.provider &&
 				(web3.provider as unknown as Web3BaseProvider).getStatus() === 'connecting'
 			) {
 				await waitForOpenConnection(web3, currentAttempt);
 			}
-			(web3.provider as unknown as Web3BaseProvider).disconnect(1000, '');
+
+			if (web3?.provider) {
+				(web3.provider as unknown as Web3BaseProvider).disconnect(1000, '');
+			}
 		}
 	});
 
+	it('should be able to create web3 object without provider', () => {
+		expect(() => new Web3()).not.toThrow();
+	});
+
+	it('should be able use functionality with web3 object not dependant on provider', () => {
+		web3 = new Web3();
+
+		expect(web3.utils.hexToNumber('0x5')).toBe(5);
```

Comment:
Seems ok, but I was thinking we should have test that goes through multiple scenarios without provider. Like we should add ABI encoding without provider :
```
  const web3 = new Web3();
  const contract = new web3.eth.Contract(ContractABI);
   await contract.methods.Func(12).encodeABI(); // and see it works
```
This will cover more scenarios, web3 instance creation, contract instance creation, encoding etc

Reply:
Such behaviour is better candidate for unit tests, not the integration. I had added one test case though in the integration tests.

Reply:
Only encoding param can be yes, but for scenarios that involve multiple packages, integration tests are more good. like for eth_call of specific function of contract without provider for offline signing etc.

Reply:
These functions pure functions, have no dependency on any related code that can effect the behavior. So these functions can be better tested in unit tests with all possible input/output. If those functions had any state which are preserved in the class then we would have gone for integration tests.